controllers/datafoundation.go

```diff
@@ -271,12 +273,21 @@ func (r *dataFoundationReconciler) reconcileStorageCluster() error {
 			return err
 		}
 
-		// Convert the desired size to the device set count based on the underlaying OSD size
-		desiredDeviceSetCount := int(math.Ceil(float64(r.spec.UsableCapacityInTiB) / templates.OSDSizeInTiB))
+		// get the desired capacity in TiB planned to be consumed by user in range 1-4TiB
+		desiredOSDSizeInTiB := math.Min(math.Ceil(float64(r.spec.UsableCapacityInTiB)), templates.VerticalScalerUpperBoundInTiB)
```

Comment:
Something does not make sense with 
```
desiredOSDSizeInTiB := math.Min(math.Ceil(float64(r.spec.UsableCapacityInTiB)), templates.VerticalScalerUpperBoundInTiB)
```

Why are we basing desired state on actual state (`r.spec.*`)?


Also `ceil(float(int))` will just return the `int` as a float. 
This means that the above statement can be simplified to
```suggestion
desiredOSDSizeInTiB := math.Min(
    float64(r.spec.UsableCapacityInTiB), 
    float64(templates.VerticalScalerUpperBoundInTiB),
)
```

Reply:
> Why are we basing desired state on actual state (r.spec.*)?

That was my exact question, but I just followed the logic (that don't make sense) that existed prior to my change [controllers/datafoundation.go#L275](https://github.com/red-hat-storage/managed-fusion-agent/blob/main/controllers/datafoundation.go#L275)
```Go
desiredDeviceSetCount := int(math.Ceil(float64(r.spec.UsableCapacityInTiB) / templates.OSDSizeInTiB))
```

Reply:
The `Ceil` on the orignal call was around the division where it made sense that we will round up the result. 
In the new code it is just a `Ceil` on a float conversion. That is the difference

Reply:
Note I was referring to the questionable use of `r.spec` and nothing else